modules/xmpp/ChatRoom.js

```diff
@@ -1837,7 +1841,9 @@ export default class ChatRoom extends Listenable {
             }
             eventEmitter.on(XMPPEvents.MUC_LEFT, onMucLeft);
             this.doLeave();
-        });
+        }));
+
+        return Promise.all(promises);
```

Comment:
Promise.all will resolve when the first one fails. I'd say we want Promise.allSettled here.

Reply:
Let's leave it as-is for now.